### ISSUE TITLE
[css-inline-3] Allow re-ordering of <text-edge> keywords

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#10713]
In line with the "re-ordering should be allowed if it's not causing parsing ambiguity" principle, maybe we should allow re-ordering of the `<text-edge>` keywords?

Current syntax for `<text-edges>` (used in `text-box-edge` and `line-fit-edge`):

```
<text-edge> = [ text | cap | ex | ideographic | ideographic-ink ]
              [ text | alphabetic | ideographic | ideographic-ink ]?
```

(Single keywords are duplicated, or default to `text` if that's not possible.)

Change to allow re-ordering:
```
<text-edge> = [ text | cap | ex | ideographic | ideographic-ink ]
              || [ text | alphabetic | ideographic | ideographic-ink ]
```

(It also allows `alphabetic` to be specified alone, which is maybe not particularly useful, but there's no particular reason to disallow it.)

**Comment by kojiishi** [comment]
Isn't
```
text-box-edge: ideographic ideographic-ink;
```
ambiguous?

**Comment by fantasai** [comment]
Yeah, I guess we'd need a priority rule, like in background-position.

Possibly it's not worth allowing reordering in this case. :)

**Comment by fantasai** [comment]
Other thought: one of the reasons I went with the `text` keyword was to match up with `text-top` and `text-bottom`. Perhaps in the two-keyword version, they can be made explicit?

A reorderable syntax could look like:

```
<text-edge> = text | ideographic | ideographic-ink |
              [ text-top | cap | ex | ideographic-top | ideographic-ink-top ]
              || [ text-bottom | alphabetic | ideographic-bottom | ideographic-ink-bottom ]
```

**Comment by bfgeek** [comment]
I'm not sure allowing reordering is worth the extra complexity for the syntax split with `ideographic-top`/`ideographic-bottom` for CJK web developers.
